crates/core/src/config/port_allocation.rs

```diff
@@ -0,0 +1,47 @@
+use std::net::UdpSocket;
+
+pub(crate) fn find_available_port() -> std::io::Result<u16> {
+    // Bind to port 0 to let OS assign an available port
+    let socket = UdpSocket::bind("127.0.0.1:0")?;
```

Comment:
Shouldnt this be binding to whatever IP address is specified in the config it the default one?

Reply:
Ah yes, it should.